design/vela-core/environment.md

```diff
@@ -1,294 +1,262 @@
-          slack: slack_url
-          dingding: dingding_url
-```
+## Considerations
 
-Under the hood, the `slo-policy` and `chaos-policy` Policies would be translated to data objects (e.g. ConfigMap
-or dedicated Policy objects) that contains policy data.
-The `check-policy` Workflow Steps would trigger `check-policy` Operator to register the apps
-to handlers of the target's policy checks, and handle notifications.
 
+## Appendix
 
-### case 3: Operator & CRD
+### Creating Kubernetes cluster
 
-In this case, an initializer defines the system Operators ands CRDs to be installed on a cluster.
-Deploying such an initializer will setup these services and definitions on the host cluster.
+In this section, we will show how to create a Kubernetes cluster on Alibaba Cloud using Terraform.
 
-Platform team would define these Operators and CRDs as Components, and deploy such an environment as an Initializer:
+We will first setup Terraform Alibab provider:
 
 ```yaml
-kind: Initializer
+apiVersion: core.oam.dev/v1beta1
+kind: Application
 metadata:
-  name: prod-env
+  name: terraform-alibaba
+  namespace: vela-system
 spec:
-  appTemplate:
-    spec:
-      components:
-        - name: prod-monitoring-operator
-          # Reuse the built-in Helm/Kustomize ComponentDefinition to deploy Operators from Git repo.
-          type: helm-git
-          properties:
-            chart:
-              repository: operator-git-repo-url
-              name: monitoring-operator
-              version: 3.2.0
-        - name: prod-logging-operator
-          type: kustomize-git
-          properties:
-            sourceRef:
-              kind: GitRepository
-              name: logging-operator
-            path: ./apps/prod
+  components:
+    - name: default
+      type: raw
+      properties:
+        apiVersion: terraform.core.oam.dev/v1beta1
+        kind: Provider
+        metadata:
+          namespace: default
+        spec:
+          provider: alibaba
+          region: cn-hongkong
+          credentials:
+            source: Secret
+            secretRef:
+              namespace: vela-system
+              name: alibaba-account-creds
+              key: credentials
+  workflow:
+  - name: wait-dependencies
+    type: depends-on-app
+    properties:
+      name: terraform
+      namespace: vela-system
+      
+  - name: apply-self
+    type: apply-application
 ```
 
-Under the hood, Vela will support built-in ComponentDefinitions to deploy Helm charts or Kustomize folders
-via git url. We can use this functionality to deploy system operators.
-
-
-### case 4: Shared Service
-
-In this case, an Initializer defines shared services to be installed on the cluster.
-Deploying such an initializer will setup these services on the host cluster.
-
-Platform team would define the shared services as Components, and deploy such an environment as an Initializer:
+Then we will create a Kubernetes cluster:
 
 ```yaml
-kind: Initializer
+apiVersion: core.oam.dev/v1beta1
```

Comment:
this demo should be fixed

```yaml
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: managed-cluster
  namespace: vela-system
spec:
  components:
    - name: ack-worker
      type: alibaba-ack
      properties:
        writeConnectionSecretToRef:
          name: ack-conn
          namespace: vela-system
  workflow:
    steps:
      - name: wait-dependencies
        type: depends-on-app
        properties:
          name: terraform-alibaba
          namespace: vela-system

      - name: wait-dependencies
        type: depends-on-app
        properties:
          name: ocm-cluster-manager
          namespace: vela-system
          
      - name: terraform-ack
        type: create-ack
        properties:
          component: ack-worker
        outputs:
          - name: connInfo
            exportKey: connInfo

      - name: register-ack
        type: register-cluster
        inputs:
          - from: connInfo
            parameterKey: connInfo
        properties:
          # user should set public network address of APIServer
          hubAPIServer: {{ public network address of APIServer }}
          env: prod
          initNameSpace: default
          patchLabels:
            purpose: test
```

Reply:
Should the second step name `wait-dependencies` be renamed?